lib/vrl/stdlib/src/parse_query_string.rs

```diff
@@ -0,0 +1,97 @@
+use url::form_urlencoded;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseQueryString;
+
+impl Function for ParseQueryString {
+    fn identifier(&self) -> &'static str {
+        "parse_query_string"
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "parse query string",
+            source: r#"parse_query_string({"query": "foo=1&bar=2"}.query)"#,
+            result: Ok(r#"
+                {
+                    "foo": "1",
+                    "bar": "2"
+                }
+            "#),
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let target = arguments.required_query("target")?;
+        Ok(Box::new(ParseQueryStringFn { target }))
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "target",
+            kind: kind::ANY,
+            required: true,
+        }]
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ParseQueryStringFn {
+    target: expression::Query,
```

Comment:
We talked on Slack that this should be a string value rather than requiring a object path to be specified. Once that change is made, we may want to rename `target` -> `value` to match other `parse_X` functions.

Reply:
Yep, misunderstood the requirements first, updated. Thanks!